metagpt/roles/role.py

```diff
@@ -151,13 +177,19 @@ async def _think(self) -> None:
             return
         prompt = self._get_prefix()
         prompt += STATE_TEMPLATE.format(history=self._rc.history, states="\n".join(self._states),
-                                        n_states=len(self._states) - 1)
+                                        n_states=len(self._states) - 1, previous_state=self._rc.state)
+        # print(prompt)
         next_state = await self._llm.aask(prompt)
         logger.debug(f"{prompt=}")
-        if not next_state.isdigit() or int(next_state) not in range(len(self._states)):
+        if not next_state.isdigit() or int(next_state) not in range(-1, len(self._states)):
             logger.warning(f'Invalid answer of state, {next_state=}')
-            next_state = "0"
-        self._set_state(int(next_state))
+            next_state = None
```

Comment:
If the LLM returns invalid response, it should have a retry and can't occupy the max_react_loop.
max_react_loop = 2, it should have `_think _act _think _act`. If the LLM returns invalid response in first _think, it will break now. But how about having a retry make the loop continue?

Reply:
我觉得先不搞这么复杂吧，可以放用LLM做think的单独优化来做。目前就直接退出